### ISSUE TITLE
Error on running `make` on macOS

### ISSUE THREAD

**Issue by HeroicHitesh** [bitcoin/bitcoin#25844]
**Expected behavior**
Bitcoin Core should compile without error as per [this guide](https://github.com/bitcoin/bitcoin/blob/master/doc/build-osx.md#2-compile)

**Actual behavior**
Getting error on running `make` command
<img width="1440" alt="make command error" src="https://user-images.githubusercontent.com/37622734/184547141-918224a7-8faa-49b6-8634-78d42a214a3d.png">

**To reproduce**

- Follow the [guide](https://github.com/bitcoin/bitcoin/blob/master/doc/build-osx.md#4-clone-bitcoin-repository) till here.
- Skip Optional Dependency installation
- Configure for No Wallet or GUI as shown [here](https://github.com/bitcoin/bitcoin/blob/master/doc/build-osx.md#no-wallet-or-gui)
- Then try compiling using `make`

**System information**

- Cloned bitcoin repo, at time of cloning latest commit id = dc9d6626835ec2864dfa747c12071dabdc95b919
- OS = macOS Big Sur
- Disk Type = SSD

**Comment by hebasto** [comment]
@HeroicHitesh 
Hmm, your build environment looks a bit weird, considering the `BITROCK_COMMON_ROOTDIR` variable from your log.

**Comment by jarolrod** [comment]
make works just fine on macOS, please recheck your user environment for any custom changes/environment variables that are preventing you from building.

**Comment by HeroicHitesh** [comment]
Hi @hebasto @jarolrod thanks for the response. I am still confused as to where to look to fix the error.
I am getting following output for `./configure --without-wallet --with-gui=no` command
<img width="1440" alt="Screenshot 1" src="https://user-images.githubusercontent.com/37622734/184566299-6c7a3f08-dcf9-49a5-b300-8a85db802297.png">
<img width="1440" alt="Screenshot 2" src="https://user-images.githubusercontent.com/37622734/184566408-0229a187-cabb-4e9b-bb99-3579f465c476.png">

<details>
<summary>And my .zshrc file looks like this</summary>

```
export LD_LIBRARY_PATH=/usr/local/lib:$LD_LIBRARY_PATH
export PKG_CONFIG_PATH=/usr/local/lib/pkgconfig:$PKG_CONFIG_PATH

# >>> DB >>>
# !! MySQL !!
export PATH="/usr/local/opt/mysql@5.7/bin:$PATH"

# !! PostgreSQL !!
export PATH=/usr/lib/postgresql/12.2/bin/:$PATH
# <<< DB <<<



# >>> NVM >>>
export NVM_DIR="$HOME/.nvm"
[ -s "$NVM_DIR/nvm.sh" ] && . "$NVM_DIR/nvm.sh" # This loads nvm
[ -s "$NVM_DIR/bash_completion" ] && . "$NVM_DIR/bash_completion" # This loads nvm bash_completion
# <<< NVM <<<



# >>> Mobile >>>
# !! Flutter !!
export PATH="$PATH:$HOME/Developer/flutter/bin"
export ANDROID_HOME="/Users/apple/Library/Android/sdk"
export PATH="$PATH:$HOME/Developer/flutter/bin:$HOME/Developer/flutter/bin/cache/dart-sdk/bin:$HOME/.pub-cache/bin"

# !! Android !!
export ANDROID_HOME="$HOME/Library/Android/sdk"
export PATH="$ANDROID_HOME/tools:$ANDROID_HOME/tools/bin:$ANDROID_HOME/platform-tools:$PATH"
# <<< Mobile <<<



# >>> Python >>>
# !! Pyenv !!
if command -v pyenv 1>/dev/null 2>&1; then
  eval "$(pyenv init -)"
fi

# !! Python !!
# # Setting PATH for Python 3 installed by brew
# export PATH=/usr/local/share/python:$PATH

# # Configuration for virtualenv
# export WORKON_HOME=$HOME/.virtualenvs
# export VIRTUALENVWRAPPER_PYTHON=/usr/local/bin/python3
# export VIRTUALENVWRAPPER_VIRTUALENV=/usr/local/bin/virtualenv
# source /usr/local/bin/virtualenvwrapper.sh

# Setting PATH for Python 3 installed by brew
export PATH=/Users/apple/.pyenv/shims/python3:$PATH

# Configuration for virtualenv
export WORKON_HOME=$HOME/.virtualenvs
export VIRTUALENVWRAPPER_PYTHON=/Users/apple/.pyenv/versions/3.8.2/bin/python3
export VIRTUALENVWRAPPER_VIRTUALENV=/Users/apple/.pyenv/versions/3.8.2/bin/virtualenv
source /Users/apple/.pyenv/versions/3.8.2/bin/virtualenvwrapper.sh
# <<< Python <<<



# >>> GitLab/GDK >>>
# PATH="/usr/local/opt/coreutils/libexec/gnubin:$PATH"
. /usr/local/share/zsh/site-functions/_asdf

export PATH="/usr/local/opt/libiconv/bin:$PATH"

export GUILE_LOAD_PATH="/usr/local/share/guile/site/3.0"
export GUILE_LOAD_COMPILED_PATH="/usr/local/lib/guile/3.0/site-ccache"
export GUILE_SYSTEM_EXTENSIONS_PATH="/usr/local/lib/guile/3.0/extensions"

export GUILE_TLS_CERTIFICATE_DIRECTORY=/usr/local/etc/gnutls/

export PKG_CONFIG_PATH="/usr/local/opt/icu4c/lib/pkgconfig:$PKG_CONFIG_PATH"

# !! Ruby !!
export PATH="$HOME/.asdf/installs/ruby/2.6.6/bin:$PATH"

export PATH="/usr/local/opt/ruby/bin:$PATH"

export PATH="$HOME/.asdf/installs/golang/1.14.6/go/bin:$PATH"
export PATH="$HOME/.asdf/installs/minio/2020-07-27T18-37-02Z/bin:$PATH"
export PATH="$HOME/.asdf/installs/redis/6.0.6/bin:$PATH"
# <<< GitLab/GDK <<<



# >>> PHP >>>
export PATH="/usr/local/Cellar/php/7.4.10/bin:$PATH"
export PATH="/usr/local/Cellar/php/7.4.10/sbin:$PATH"
alias "composer"="php /usr/local/Cellar/composer/2.1.3/bin/composer"
export PATH="/Applications/XAMPP/xamppfiles/bin:$PATH"

# !! Joomla !!
alias jclean="rm -rf administrator/templates/atum/css; rm -rf templates/cassiopeia/css; rm -rf administrator/templates/system/css; rm -rf templates/system/css; rm -rf media/; rm -rf node_modules/; rm -rf libraries/vendor/;rm -f administrator/cache/autoload_psr4.php;rm -rf installation/template/css"
alias jinstall="jclean; composer install; npm ci"
# <<< PHP <<<



# >>> Java >>>
export JAVA_HOME=$(/usr/libexec/java_home)
jdk() {
      version=$1
      unset JAVA_HOME;
      export JAVA_HOME=$(/usr/libexec/java_home -v"$version");
      java -version
}
# <<< Java <<<


# >>> Rust >>>
. "$HOME/.cargo/env"
# <<< Rust <<<



# >>> Cloud >>>

# !! kubectl !!
[[ /usr/local/bin/kubectl ]] && source <(kubectl completion zsh)
# <<< Cloud <<<


export LD_LIBRARY_PATH=/usr/local/lib:$LD_LIBRARY_PATH
export PKG_CONFIG_PATH=/usr/local/lib/pkgconfig:$PKG_CONFIG_PATH
```
</details>

Also, while running `make` command for the first time I got a lot of warnings like this
<img width="1440" alt="Warnings" src="https://user-images.githubusercontent.com/37622734/184567495-cbb20991-e764-409e-8130-af55ec27eec3.png">
I am not sure if they caused the issue or not, so just sharing for the reference

Kindly let me know if there's any other setting or file where I can look for to fix this issue.

**Comment by hebasto** [comment]
> Also, while running make command for the first time I got a lot of warnings...

To silence them add --enable-suppress-external-warnings to your configure invocation.

**Comment by fanquake** [comment]
It's not exactly clear what the problem is here, however it's certainly a local issue. Feel free to reopen if you can recreate the issue in a clean environment. The `BITROCK_COMMON_ROOTDIR` problem is nothing to with Bitcoin Core or it's build instructions.

**Comment by HeroicHitesh** [comment]
Thanks @hebasto for the tip, it helped a lot.

Ok @fanquake I actually thought `BITROCK_COMMON_ROOTDIR` issue was something related to Bitcoin Core, since this is the first time I faced it.
But after your comment, I did some online search, and thankfully was able to fix it by following [this comment](https://github.com/mongodb/mongo-php-driver/issues/883#issuecomment-406707863).

Thanks a lot for the support everyone.
